src/Swift.Bindings/src/Model/MethodDecl.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace BindingsGeneration
+{
+    /// <summary>
+    /// Represents a method declaration.
+    /// </summary>
+    public sealed record MethodDecl
+    {
+        /// <summary>
+        /// Name of the method.
+        /// </summary>
+        public required string Name { get; init; }
+        
+        /// <summary>
+        /// Mangled name of the method.
+        /// </summary>
+        public required string MangledName { get; init; }
+        
+        /// <summary>
+        /// Signature of the method.
+        /// </summary>
+        public required IEnumerable<TypeDecl> Signature { get; set; }
```

Comment:
I wonder if having signature represented just by a simple `IEnumerable<TypeDecl>` will be enough in the future or if it would be better to declare `SignatureDecl` and split return and params into separate fields. 

My concerns here is the possibility of implicit `self`/`this` but I'm unsure if that will concern us in the future. Nonetheless, for the prototype we can keep this for simplicity.

Reply:
It's a good idea, and we can consider implementing it in the future. Swift ABI signature is presented in this way (return type + params).